sdk/servicebus/azure-servicebus/CHANGELOG.md

```diff
@@ -10,6 +10,20 @@
   - `ServiceBusSession`: `get_state`, `set_state` and `renew_lock`
   - `ReceivedMessage`: `renew_lock`
 
+**Breaking Changes**
+
+* `azure.servicebus.ServiceBusError` now inherits from `azure.core.exception.AzureError`.
+* The parameter `inner_exception` that `ServiceBusError` takes is now renamed to `error`.
```

Comment:
emmm, I'm following the `AzureError` in azure-core -- https://github.com/Azure/azure-sdk-for-python/blob/28a99a0fab81b336b334fbdf032462bb77e67b36/sdk/core/azure-core/azure/core/exceptions.py#L187-L218

I don't think `__cause__` is mentioned in our guideline: https://azure.github.io/azure-sdk/python_implementation.html#error-handling, so I would assume azure-core as the guideline for error definition. So to me, your question seems to be a guideline-scope one.

btw,  by "this should both populate and be associated with __cause__ instead", do you mean the code being this:
```
class ServiceBusError:
    def __init__(self, ..., error=None, inner_exception=None):
        pass
    def __cause__(self):
        return self.inner_exception
```

Reply:
more context: `__cause__` introduced in python3, python2.7 doesn't support `raise xxx from yyy`.
see the pep3134 here: https://www.python.org/dev/peps/pep-3134/

Reply:
Aha, I should have looked deeper into AzureError (since I knew we used inner in our other classes, but didn't realize it was influenced from there), good callout.

So I'd revise my initial statement to "properly support cause" instead of "rename to realign" as well, for which your sample at the bottom is exactly what I had in mind.  From our OOB discussion, you're driving the alignment with the architects to reconcile this and __cause__ so I'll let you close on that and wait for your ping on when you need the rubber stamp++ 👍

Reply:
opened issue for considering adding `__cause__` support into azure-core:
https://github.com/Azure/azure-sdk-for-python/issues/14607